src/modules/keyboardmanager/dll/KeyboardEventHandlers.cpp

```diff
@@ -598,6 +636,42 @@ namespace KeyboardEventHandlers
                                 i++;
                             }
 
+                            it->second.isShortcutInvoked = false;
+                            it->second.winKeyInvoked = ModifierKey::Disabled;
+                            // If app specific shortcut has finished invoking, reset the target application
+                            if (activatedApp != KeyboardManagerConstants::NoActivatedApp)
+                            {
+                                keyboardManagerState.SetActivatedApp(KeyboardManagerConstants::NoActivatedApp);
+                            }
+                            lock.unlock();
+                            UINT res = ii.SendVirtualInput((UINT)key_count, keyEventList, sizeof(INPUT));
+                            delete[] keyEventList;
+                            return 1;
+                        }
+                        else if (std::get<DWORD>(it->second.targetShortcut) == CommonSharedConstants::VK_DISABLED)
+                        {
+                            // Key down for original shortcut modifiers and action key, dummy key, and current key press
+                            size_t key_count = src_size + 1 + 1;
+
+                            LPINPUT keyEventList = new INPUT[key_count]();
+                            memset(keyEventList, 0, sizeof(keyEventList));
+
+                            // Set old shortcut key down state
+                            int i = 0;
+                            KeyboardManagerHelper::SetModifierKeyEvents(it->first, it->second.winKeyInvoked, keyEventList, i, true, KeyboardManagerConstants::KEYBOARDMANAGER_SHORTCUT_FLAG);
+
+                            // Set old action key
```

Comment:
[nit] maybe just add a comment mentioning that the action key will definitely be pressed down (since we reset the invoked state for shortcut to key remaps if the action key is released). Just so that if anyone else looks at this code they have context for why this is different from the case of shortcut to shortcut, where we check if the action key is pressed or not.